web-components/src/components/form/BatchBasicsForm.tsx

```diff
@@ -0,0 +1,243 @@
+import { Box, IconButton, Link } from '@mui/material';
+import { useTheme, DefaultTheme as Theme } from '@mui/styles';
+import { string, date, object, AnyObjectSchema } from 'yup';
+
+import { Body } from '../typography';
+import InputLabel from '../inputs/InputLabel';
+import OnBoardingCard from '../cards/OnBoardingCard';
+import TextField from '../inputs/TextField';
+import ControlledTextField from '../inputs/ControlledTextField';
+import SelectTextField from '../inputs/SelectTextField';
+import { DatePickField } from '../inputs/DatePickField';
+import SaveFooter from '../fixed-footer/SaveFooter';
+import OutlinedButton from '../buttons/OutlinedButton';
+import TrashIcon from '../icons/TrashIcon';
+import { AddCertificationModal } from '../modal/AddCertificationModal';
+import {
+  requiredMessage,
+  invalidDate,
+  vcsRetirementSerialRE,
+  invalidVCSRetirement,
+  invalidVCSID,
+  numericOnlyRE,
+} from '../inputs/validation';
+import { NameUrl } from '../../types/rdf';
+import { BatchMetadataLD } from '../../types/rdf/batch-metadata';
+
+export interface BatchBasicsFormValues {
+  class_id: string;
+  start_date: Date | null;
+  end_date: Date | null;
+  metadata?: Partial<BatchMetadataLD>;
+}
+
+const vcsMetadataSchema: AnyObjectSchema = object({
+  'regen:vcsProjectId': string()
+    .matches(numericOnlyRE, { message: invalidVCSID })
+    .required(requiredMessage),
+  'regen:vcsRetirementSerialNumber': string()
+    .matches(vcsRetirementSerialRE, { message: invalidVCSRetirement })
+    .required(requiredMessage),
+});
+
+const validationSchema = object({
+  class_id: string().required(requiredMessage),
+  start_date: date().required(requiredMessage).typeError(invalidDate),
+  end_date: date().required(requiredMessage).typeError(invalidDate),
+  metadata: vcsMetadataSchema,
+});
+
+const BatchBasicsForm: React.FC<{
+  submit: (values: BatchBasicsFormValues) => Promise<void>;
+}> = ({ submit }) => {
+  return (
+    <Formik
+      enableReinitialize
+      validateOnMount
+      initialValues={{
+        class_id: '',
+        start_date: null,
+        end_date: null,
+        metadata: undefined,
+      }}
+      validationSchema={validationSchema}
+      onSubmit={async (values, { setSubmitting }) => {
+        setSubmitting(true);
+        try {
+          await submit(values);
+          setSubmitting(false);
+        } catch (e) {
+          setSubmitting(false);
+        }
+      }}
+    >
+      {({ submitForm, isValid, isSubmitting, touched, values }) => {
+        const isVCS = values?.class_id === 'C01';
+
+        return (
+          <Form>
+            <OnBoardingCard>
+              <Field
+                name="class_id"
+                label="Credit Class"
+                required
+                component={SelectTextField}
+                options={[
+                  { value: '', label: 'Choose Credit Class' },
+                  { value: 'C01', label: 'Verified Carbon Standard (C01)' },
+                ]}
```

Comment:
this shouldn't be hardcoded, we should fetch from on-chain credit classes

Reply:
in-progress

Reply:
agreed. done.